app/screens/settings/notification_auto_responder/notification_auto_responder.tsx

```diff
@@ -85,20 +93,68 @@ const NotificationAutoResponder = ({currentUser}: NotificationAutoResponderProps
 
     const styles = getStyleSheet(theme);
 
+    const close = () => popTopScreen(componentId);
+
+    const saveButton = useMemo(() => {
+        return {
+            id: SAVE_OOO_BUTTON_ID,
+            enabled: false,
+            showAsAction: 'always' as const,
+            testID: 'notification_settings.auto_res.save.button',
+            color: theme.sidebarHeaderTextColor,
+            text: intl.formatMessage({id: 'settings.save', defaultMessage: 'Save'}),
+        };
+    }, [theme.sidebarHeaderTextColor]);
+
+    const canSave = useCallback((enabled: boolean) => {
+        const buttons = {
+            rightButtons: [{
+                ...saveButton,
+                enabled,
+            }],
+        };
+        setButtons(componentId, buttons);
+    }, [componentId, saveButton]);
+
+    const shouldSaveAutoResponder = useCallback((notify: Partial<UserNotifyProps>) => {
+        const {auto_responder_active: autoResponderActive} = notify;
+        const prevProps = getNotificationProps(currentUser);
+        const updatedMsg = prevProps.auto_responder_message !== notify.auto_responder_message;
+        const enabling = currentUser.status !== General.OUT_OF_OFFICE && autoResponderActive === 'true';
+        const disabling = currentUser.status === General.OUT_OF_OFFICE && autoResponderActive === 'false';
+
+        canSave(enabling || disabling || updatedMsg);
+    }, [currentUser.status, canSave]);
+
+    const updateNotifyProps = useCallback((obj: Partial<UserNotifyProps>) => {
+        const notify = {...notifyProps, ...obj};
+        setNotifyProps(notify);
+        shouldSaveAutoResponder(notify);
+    }, [shouldSaveAutoResponder]);
+
     const onAutoResponseToggle = useCallback((active: boolean) => {
-        setNotifyProps({
-            ...notifyProps,
-            auto_responder_active: `${active}`,
-        });
+        updateNotifyProps({auto_responder_active: `${active}`});
```

Comment:
This shouldn't work.

updateNotifyProps only depends on `shouldSaveAutoResponder`. `onAutoResponderToggle` depends on notifyProps, but any change on notifyProps won't change anything, since `updateNotifyProps` won't be updated.

I would depend on these functions on `updateNotifyProps` instead, and let the correct dependencies be handled by `updateNotifyProps`.